include/dkm.hpp

```diff
@@ -121,7 +125,12 @@ std::vector<std::array<T, N>> random_plusplus(const std::vector<std::array<T, N>
 		input_size_t i = 0;
 		std::discrete_distribution<input_size_t> generator(distances.size(), 0.0, 0.0, [&distances, &i](double) { return distances[i++]; });
 #endif
-		means.push_back(data[generator(rand_engine)]);
+        auto index = generator(rand_engine);
+        if (index == generator.probabilities().size())
+        {
+            index = 0;
+        }
+        means.push_back(data[index]);
```

Comment:
what happens if size == 0?

Reply:
actually found it, this is case 2 here  https://en.cppreference.com/w/cpp/numeric/random/discrete_distribution/discrete_distribution
so we should be good